drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -132,6 +133,27 @@ class DragAndDropBlock(
         default=True,
         enforce_type=True,
     )
+    showanswer = String(
```

Comment:
`showanswer` is a global course setting, you can go to "Settings" -> "Advanced Settings" in the studio and search for "Show Answer".

Reply:
@abodacs, we are defining it in the XBlock, so it's an XBlock-specific option which **can be** overridden in the "Advanced Settings", as I've mentioned [in this comment](https://github.com/openedx/xblock-drag-and-drop-v2/pull/294#discussion_r983343560). You can find an explanation of how the Scope.settings works [in this document](https://edx.readthedocs.io/projects/xblock/en/latest/fields.html#xblock.fields.Scope).

Reply:
@abodacs, I'm answering your Jira comment here, to preserve the context - yes, your understanding is correct. When you add the `showanswer` field to an XBlock, it determines the value in the following order:

1. If the value is set in the DnDv2 editor - use it. This is an XBlock-specific setting.
2. If the XBlock-specific setting is not set - use the one from the course "Advanced Settings".

This mechanism is already implemented, so we don't need to worry about it. The only thing we want is to allow defining it in the Studio editor. Therefore, we want to add a dropdown with possible values (like [this one](https://github.com/openedx/xblock-drag-and-drop-v2/blob/cfd2b2dcc7e77e3fbb8949f8f1d8a548b29aa078/drag_and_drop_v2/templates/html/drag_and_drop_edit.html#L25-L34)).
There is a small catch - this XBlock doesn't have a native "reset" value option, but we need to allow course authors to reset this value in case they want to revert to using a course-specific option. I suggest adding a ["Clear" button](https://github.com/open-craft/edx-platform/blob/740d6ed59c7bb4821c23db936b0b4402f22d22d7/cms/templates/js/metadata-option-entry.underscore#L12-L14) which sets an empty value for this setting. Then, we can handle it [here](https://github.com/openedx/xblock-drag-and-drop-v2/blob/cfd2b2dcc7e77e3fbb8949f8f1d8a548b29aa078/drag_and_drop_v2/drag_and_drop_v2.py#L474) - when the `self.showanswer` is empty, we run `del self.showanswer`, which simply removes the XBlock-specific setting.
If you're curious about how it works, you can find it [here](https://github.com/openedx/XBlock/blob/a89dcae5b97e1822ceedbf5f4e299bfad5f3aa1a/xblock/fields.py#L550-L576).